src/analyses/mallocFresh.ml

```diff
@@ -42,7 +42,8 @@ struct
     match desc.special args with
     | Malloc _
     | Calloc _
-    | Realloc _ ->
+    | Realloc _
+    | Alloca _ ->
```

Comment:
For `realloc`:

> The original pointer `ptr` is invalidated and any access to it is undefined behavior (even if reallocation was in-place).

So even if any other thread has the original pointer, they may not use it anymore. Only the returned pointer may be used and that's still local to the allocating thread.

UAF analysis should catch such pointer uses that may happen after another thread has `realloc`-ed it because it should be assuming the worst case anyway (the memory actually being freed). Even if it actually isn't, it's still UB just like UAF.
I think this rule exists to allow consistent reasoning about the UB behavior of `realloc`.

Reply:
✅ on the realloc.
I think the `onstack=false` still needs to be fixed for `alloca` though.

Reply:
Should be ok now.